scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -45,6 +45,10 @@ class ScalametaParser(input: Input)(implicit dialect: Dialect) { parser =>
     def isInside() = nested > 0
   }
 
+  private object QuotedPatternContext {
+    var inPattern: Boolean = false
```

Comment:
i'd just copy the contents of splice context above, as-is.

or even:
```
private abstract class QuoteContextBase {
  private var nested = 0
  def enter() = nested += 1
  def exit() = nested -= 1
  def isInside() = nested > 0
}
private object QuoteSpliceContext extends QuoteContextBase
private object QuotedPatternContext extends QuoteContextBase
```

Reply:
Done.